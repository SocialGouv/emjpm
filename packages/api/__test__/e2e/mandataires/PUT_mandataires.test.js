//

const request = require("supertest");

// Fake env
process.env.SMTP_FROM = "ne-pas-repondre@emjpm.gouv.fr";
process.env.APP_URL = "https://emjpm.gouv.fr";

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

//

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

test("mandataire should update his profile", async () => {
  const token = await getTokenByUserType("mandataire");
  const newProfile = {
    email: "test@email.com",
    telephone: "phone-test"
  };

  const response = await request(server)
    .put(`/api/v1/mandataires/1`)
    .set("Authorization", "Bearer " + token)
    .send(newProfile);

  const newProfileRow = await knex("mandataires")
    .where({ id: 1 })
    .first();

  const newUserRow = await knex("users")
    .where({ id: 1 })
    .first();

  expect(newProfileRow.telephone).toBe(newProfile.telephone);
  expect(newUserRow.email).toBe(newProfile.email);

  expect(response.body).toMatchSnapshot({
    created_at: expect.any(String),
    date_mesure_update: expect.any(String)
  });
  expect(response.status).toBe(200);
});
