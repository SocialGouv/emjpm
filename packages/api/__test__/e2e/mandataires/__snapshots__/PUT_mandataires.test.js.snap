// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mandataire should update his profile 1`] = `
Object {
  "adresse": "21 rue de houx",
  "code_postal": "62000",
  "created_at": Any<String>,
  "date_mesure_update": null,
  "department_id": 6,
  "dispo_max": 5,
  "email": "test@email.com",
  "email_send": null,
  "etablissement": "UDAHF",
  "genre": null,
  "id": 1,
  "latitude": 50.5333,
  "longitude": 2.6333,
  "mesures_en_attente": 0,
  "mesures_en_cours": 1,
  "nb_secretariat": 2,
  "nom": null,
  "prenom": "Adrien",
  "secretariat": false,
  "siret": null,
  "telephone": "phone-test",
  "telephone_portable": "0101010101",
  "type": "individuel",
  "user_id": 1,
  "ville": "Arras",
  "zip": null,
}
`;
